src/plugins/console/public/application/containers/editor/legacy/console_editor/editor_output.tsx

```diff
@@ -55,6 +56,63 @@ function modeForContentType(contentType?: string) {
   return 'ace/mode/text';
 }
 
+interface Output {
+  [key: string]: {};
+}
+
+function parseResult(response: VectorTile) {
+  const data = response.layers;
+  const output: Output = {};
+
+  for (const property in data) {
+    if (data.hasOwnProperty(property)) {
+      const propertyObject = data[property];
+      const featuresArray = [];
+
+      for (let index = 0; index < propertyObject.length; index++) {
+        const feature = propertyObject.feature(index);
+        const properties = feature.properties;
+        const geometry = feature.loadGeometry()[0];
+        const typeName = VectorTileFeature.types[feature.type];
+        let coordinates = [];
+
+        const coordinatesArray = [];
+        for (const value of geometry) {
+          coordinatesArray.push([value.x, value.y]);
+        }
+
+        switch (feature.type) {
+          case 1:
+            coordinates.push(geometry[0].x, geometry[0].y);
+            break;
+          case 2: {
+            coordinates = coordinatesArray;
+            break;
+          }
+          case 3: {
+            coordinates.push(coordinatesArray);
+            break;
+          }
+        }
+
+        featuresArray.push({
+          geometry: {
+            type: typeName,
+            coordinates,
+          },
+          properties,
+        });
+      }
+
+      output[property] = [...featuresArray];
+    }
+  }
+
+  const sortedOutput = Object.fromEntries(Object.entries(output).sort().reverse()); // "meta" layer is now in top of the result
+
+  return JSON.stringify(sortedOutput, null, '\t');
+}
```

Comment:
Would it make sense to add some unit tests to verify the behavior of this function? I can see this being useful in case someone decides to refactor it. It could guard against some accidental regressions.

Also, instead of increasing the scope of this file with this logic, I'd love to see it extracted into its own folder. This way we can colocate the logic with the tests without cluttering the `console_editor` directory. So I'd suggest creating a `console_editor/mapbox_vector_tile/` directory, which would contain these files:

```
convert_mapbox_vector_tile_to_json.ts // exports the function you've added
convert_mapbox_vector_tile_to_json.test.ts // unit tests
index.ts // re-exports the function for consumption by editor_output.tsx
```